src/test/tests_common.h

```diff
@@ -343,23 +343,39 @@ tc_cli_start_basic(char *local_group_name, char *srv_group_name,
 		      CRT_FLAG_BIT_PMIX_DISABLE | CRT_FLAG_BIT_LM_DISABLE);
 	D_ASSERTF(rc == 0, "crt_init() failed; rc=%d\n", rc);
 
-	rc = crt_group_view_create(srv_group_name, grp);
-	if (!*grp || rc != 0) {
-		D_ERROR("Failed to create group view; rc=%d\n", rc);
-		assert(0);
-	}
-
 	rc = crt_context_create(crt_ctx);
 	D_ASSERTF(rc == 0, "crt_context_create() failed; rc=%d\n", rc);
 
 	rc = pthread_create(progress_thread, NULL, tc_progress_fn, crt_ctx);
 	D_ASSERTF(rc == 0, "pthread_create() failed; rc=%d\n", rc);
 
-	grp_cfg_file = getenv("CRT_L_GRP_CFG");
+	if (save_cfg) {
+		/* try until success to avoid intermittent failuresxi
+		 * under valgrind.
```

Comment:
What is the issue that you are seeing under valgrind? 
The reason why in other places we used loop with retries was due to client being launched asynchronously with servers, so until server populated group attach file client would just loop.

Now that we have wait_for_ranks() we should not see any failures during attach (once wait succeeds), so i dont think this loop should be there at all

Reply:
hmm.... the attached failed when the loop is removed along with the sleep.
i'm going to replace this loop with the version with num retries defined, to be consistent with other test cases that uses attach (original ctl, iv_client).
if the attach is expected to succeed on first try here without sleep, i think we'll need a ticket to investigate issue. once resolved, we can remove the loop.